### DISCUSSION TITLE
你们公共弹窗代码如何写的呀。在App.ts文件中不生效。

### DISCUSSION THREAD

**Discussion by codedart2018** [NervJS/taro#13663] [Q & A] [answered]
### 这个特性解决了什么问题？
不用每个页面都去写一堆代码
现在是我把公共弹窗或更多代码放在app文件中去处理。但实际它都不渲染。但在react正常的写法来说的话，它是应该渲染的。这里直接给干没了

### 这个 API 长什么样？
不要把顶层的children内容都给干掉了，正常的React写法是可以生效的！

<img width="954" alt="image" src="https://user-images.githubusercontent.com/12706830/161555456-7c28fda2-175e-4ffc-a300-de9a9c686a45.png">

**Comment by Binbiubiubiu** [comment]
可以每个页面放一个Toast组件,然后写个方法通过EventChannel 去和组件通信,参考vant

**Comment by TheKonka** [comment, marked as answer]
https://docs.taro.zone/docs/react-entry
在入口组件不会渲染任何内容，但我们可以在这里做类似于状态管理的事情

**Comment by codedart2018** [comment]
> https://docs.taro.zone/docs/react-entry 在入口组件不会渲染任何内容，但我们可以在这里做类似于状态管理的事情

不知道入口children传下去有何副作用。react是正常可以把children传递下去的。现在每个页面都要重复引入公共的，有点不优雅！期望是的能在app即做全局数据。也希望把定义的公共children给传递下去。

**Comment by laolaolulu** [comment]
我也是没法,Taro.showToast使劲造

**Comment by mattjin** [comment]
+1
